azurerm/resource_arm_hd_insight_clusters.go

```diff
@@ -0,0 +1,918 @@
+									"id": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"subnet": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"worker_node": {
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"linux_os_profile": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"password": {
+										Optional: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"ssh_keys": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeList,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"key_data": {
+													Required: true,
+													ForceNew: true,
+													Type:     schema.TypeString,
+												},
+											},
+										},
+									},
+									"username": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+								},
+							},
+						},
+						"min_instance_count": {
+							Optional:     true,
+							Default:      1,
+							ForceNew:     true,
+							Type:         schema.TypeInt,
+							ValidateFunc: validation.IntBetween(1, 32),
+						},
+						"script_actions": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"parameters": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"uri": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+								},
+							},
+						},
+						"target_instance_count": {
+							Required:     true,
+							ForceNew:     true,
+							Type:         schema.TypeInt,
+							ValidateFunc: validation.IntBetween(1, 32),
+						},
+						"vm_size": {
+							Required: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+						"vnet_profile": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
+									"subnet": {
+										Required: true,
+										ForceNew: true,
+										Type:     schema.TypeString,
+									},
```

Comment:
as above - let's instead pass in the `subnet_id` and parse the components out that we need. this means we can also fail with a more helpful error message should the resources not be found